File: templates/nodejs-service/service/src/components/example/serviceGreeting.ts
export function greet(name: string): string {
    if (name === "") {
        return "Hello world !!!";
    }
    return `Hello ${name}`;
}

export function greetEveryone(names: string[]): string {
    if (names.length === 0) {
        return "Hello everyone !!!";
    }
    const joinedNames = names.join(", ");
    return `Hello ${joinedNames}, and everyone`;
}

File: templates/go-service/service/components/example/serviceGreeting.go
package example

import (
	"fmt"
	"strings"
)

// Greet greet one person, or the world
func Greet(name string) string {
	if name == "" {
		return "Hello world !!!"
	}
	return fmt.Sprintf("Hello %s", name)
}

// GreetEveryone greet many persons, or everyone
func GreetEveryone(names []string) string {
	if len(names) == 0 {
		return "Hello everyone !!!"
	}
	return fmt.Sprintf("Hello %s, and everyone", strings.Join(names, ", "))
}
